scale-tests/batch_test.py

```diff
@@ -73,15 +78,42 @@ def _is_strict(dispatcher: typing.Dict) -> bool:
     return dispatcher["service"].get("service_account") is not None
 
 
-def submit_job(dispatcher: typing.Dict, duration: int, config: typing.List[str]):
+def _get_gpu_user_conf(args):
+    def _verify_required_args():
+        if not (args["--spark-mesos-max-gpus"] and
+                args["--spark-mesos-executor-gpus"] and
+                args["--docker-image"]):
+            log.error("""
+            Missing required arguments for running gpu jobs. Please include:
+            --spark-mesos-max-gpus
+            --spark-mesos-executor-gpus
+            --docker-image
+            """)
+
+    _verify_required_args()
+
+    # Based on testing, 20gb per GPU is needed to run the job successfully.
+    # This is due to memory being divvied up and allocated to each GPU device.
+    memory_multiplier = 20
+    memory = int(args["--spark-mesos-executor-gpus"]) * memory_multiplier
+    return ["--conf", "spark.driver.memory={}g".format(str(memory)),
```

Comment:
@samvantran  Looks like there's a problem: the submit_job() method actually hardcodes the driver memory to 2g.

Also, do you set the driver cpus? In the quota spreadsheet, I noticed it was set to 16 cpus but I don't see it here.

Reply:
Good catch! Fixed

Reply:
Is the same amount of memory requires per driver as per executor?

Reply:
they don't have to be exactly the same but in my testing in the past, the jobs would fail if the memory was less than 20gb since the docker image itself is very large.

Reply:
@susanxhuynh , you're right. I've never set `spark.driver.cores`. We could probably lower the driver cpu quota a bunch.. but the executors do need 16